### ISSUE TITLE
Comments in Binja cause high-cpu and redundant commits

### ISSUE THREAD

**Issue by VisualEhrmanntraut** [binsync/binsync#355]
### Description

Adding a comment causes BinSync to have an aneurysm. Binary Ninja uses 660% CPU. After it stops, BinSync will start committing empty comments, 10 commits at a time. 

### Steps to reproduce the bug

Add a comment inside a function

### Environment

Binary Ninja 4.1.5014-dev Personal (349de5a9)

**Comment by mahaloz** [comment]
Hi @VisualEhrmanntraut would you be able to give a little more context?

- what OS?
- what program file type?
- are you able to share a binary to make reproducibility easier?

**Comment by VisualEhrmanntraut** [comment]
Hello @mahaloz 

I am running macOS 14.4.1 (23E224). The platform of the band is windows-kernel-x86_64 (so a PE executable). The binary comes from the Windows AMD GPU drivers, and is around 100MB, so I can't really share it.

I just went to a function, and wanted to comment on an assignment (HLIL) what the field might be, for future reference. As soon as I made the comment, Binary Ninja jumped to 660% CPU and froze. After around a minute, it unfroze and BinSync was put in an infinite loop of updating an empty comment at the address since then. I managed to stop it by closing BN and removing the comment after reopening it, then connecting back to the BinSync project.

The driver in question is AMD Adrenaline 24.3.1. The BinSync project is at https://github.com/ChefKissRE/Adrenaline_24.3.1-amdkmdag.sys. if you look at the binsync/visual branch you can see the commit spam mentioned.

**Comment by mahaloz** [comment]
Thank you! This is very helpful that you have a publicly referencable project. Is this commit the start of the useless commits:
https://github.com/ChefKissRE/Adrenaline_24.3.1-amdkmdag.sys/commit/d29b0022e385f8e46b62793996f2de6075241997

I.e., is it always a comment of `0`? Were you able to trigger this by changing anything else?

**Comment by VisualEhrmanntraut** [comment]
@mahaloz It always does the same thing (seemingly random addresses, commit spam, len 0), no matter the contents of the comment or which view (disassembly, HLIL, etc) I add it from.

**Comment by mahaloz** [comment]
Ok, it's likely coming from some logic bug here:
https://github.com/binsync/libbs/blob/e27f00ead15327b0d2119c49ab94edaca7f30c9c/libbs/decompilers/binja/hooks.py#L46

I may not have expected binja to start reporting empty comments, which seems to be happening here.

**Comment by mahaloz** [comment]
Ok, it should be fixed. I reopened to allow you to close to confirm in version `4.0.3`

**Comment by VisualEhrmanntraut** [comment]
Err, I've installed it using the plugin manager. How do I update to this version before it's available there?

**Comment by mahaloz** [comment]
Yikers. You can uninstall in the Plugin manager, then do the normal install:
https://binsync.net/docs/binja/

TL;DR:
1. Uninstall plugin manager BinSync
2. Make sure binja is using local Python interpreter (not its bundled one)
3. `pip3 install binsync && binsync --install`

**Comment by VisualEhrmanntraut** [comment]
<img width="1296" alt="Screenshot 2024-03-30 at 23 17 13" src="https://github.com/binsync/binsync/assets/30368284/43db3238-a82e-4de2-8f43-69468c064c3b">

**Comment by mahaloz** [comment]
@VisualEhrmanntraut I see how that could be confusing, the `Plugins Path` is the path to your `Binja` plugins folder.

**Comment by VisualEhrmanntraut** [comment]
Couldn't fix it. I just ran
```python
pip.main(["--isolated", "--disable-pip-version-check", "install", "-U", "-t", "/Users/visual/Library/Application\ Support/Binary\ Ninja/python310", "binsync"])
```
in BN's console

**Comment by mahaloz** [comment]
Lol, that's a little wild. So you got just the binsync library to update by running that in the BN console? You can verify with `import binsync; binsync.__version__` in the console to check if you like.

**Comment by VisualEhrmanntraut** [comment]
Yes.

<img width="204" alt="Screenshot 2024-03-30 at 23 56 05" src="https://github.com/binsync/binsync/assets/30368284/8f3ed08c-24d2-4d59-9891-600c0d8d8ebd">

**Comment by mahaloz** [comment]
Amazing. Nice trick.

**Comment by VisualEhrmanntraut** [comment]
@mahaloz Tested it. The issue does not happen, but no comment is pushed to the repo when I add a comment; nothing happens. Also, if I force push the function I get the following in the log:
```
[Default] Traceback (most recent call last):
[Default]   File "/Users/visual/Library/Application Support/Binary Ninja/python310/site-packages/binsync/ui/control_panel.py", line 33, in <lambda>
[Default]     menu.addAction(f"Force Push {ctx_name}@{hex(ctx_addr)}", lambda : self.controller.force_push_function(ctx_addr))
[Default] AttributeError: 'BSController' object has no attribute 'force_push_function'. Did you mean: 'force_push_functions'?
```
But I assume that's not related to this.

**Comment by mahaloz** [comment]
Ok, the force push is a different failure. You said the comment is not pushed, but was it committed? I.e if you git log locally is it there and just failed to be pushed to the remote?

**Comment by VisualEhrmanntraut** [comment]
It's not pushing nor committing anything.

**Comment by VisualEhrmanntraut** [comment]
Guess it was stuck in a bad state. Now it works.
